crates/orchestrator/src/data_storage/aws_s3/config.rs

```diff
@@ -12,6 +12,9 @@ pub struct AWSS3Config {
     pub s3_bucket_name: String,
     /// S3 Bucket region
     pub s3_bucket_region: String,
+    /// Endpoint url
+    #[cfg(test)]
+    pub endpoint_url: String,
```

Comment:
Why is this only used in tests? Is there no equivalent notion when connecting to AWS directly? It would be better to avoid this difference between production and testing, if possible.

Reply:
Localstack endpoint URL : http://localhost.localstack.cloud:4566
AWS endpoint URL. : https://s3.{region}.amazonaws.com/{bucket-name}/{key}

We are using [localstack](https://hub.docker.com/r/localstack/localstack) for mocking AWS locally.
Localstack requires `endpoint_url` to get connected.

Connection to AWS doesn't necessarily require the endpoint url as shown [here](https://docs.rs/aws-config/latest/aws_config/index.html).
Rather, the connection to localstack does as shown [here](https://docs.aws.amazon.com/sdk-for-rust/latest/dg/localstack.html#:~:text=if%20use_localstack()%20%7B%0A%20%20%20%20%20%20%20%20shared_config%20%3D%20shared_config.endpoint_url(LOCALSTACK_ENDPOINT)%3B%0A%20%20%20%20%7D%3B).

Hence this implementation is under the tests flag.
What do you propose as a solution?, 
Do you think changing prod code as how tests are written (adding endpoint url on prod just because it's available on tests) is a valid way to go ?

Reply:
But specifying the endpoint is still a possibility on AWS, right? I'm not in favor of "modifying production code", but here it's just that you have 2 different ways of connecting: either you specify the region, bucket key etc or you provide the endpoint directly. It turns out that localstack only supports the endpoint method, sure. 

In effect you have two usable connection solutions => you could use an enumeration to support both features, for example giving priority to the endpoint method if someone wants to hardcode it for some reason and then support the default method as a backup. This way, your code is the same for testing vs prod and you support two methods to connect to AWS. WDYT?

Reply:
Hey @odesenfans 
I agree with your implementation and it is more cleaner. I will implement this and push the code.